### ISSUE TITLE
Incorrect predictions of Mobilenet_V2

### ISSUE THREAD

**Issue by JionBin** [tensorflow/tensorflow#31229]
Hi,
I built the tensorflow lite on iMX8(4xA53) platform. I can get correct results when using models of mobilenet_v1_1.0_224, mobilenet_v1_1.0_224_quant and mobilenet_v2_1.0_224 as input model for  [label_image](https://github.com/tensorflow/tensorflow/tree/v2.0.0-alpha0/tensorflow/lite/examples/label_image). But I got unreasonable predictons when using [mobilenet_v2_1.0_quant](https://storage.googleapis.com/download.tensorflow.org/models/tflite_11_05_08/mobilenet_v2_1.0_224_quant.tgzl).

Here are the results:
_./label_image -i dog.bmp -m **mobilenet_v1_1.0_224.tflite** -l labels.txt                          
Loaded model mobilenet_v1_1.0_224.tflite
resolved reporter
invoked 
average time: 200.31 ms 
0.987781: 209 Labrador retriever
0.00432148: 208 golden retriever
0.00298653: 163 beagle
0.00134322: 160 Rhodesian ridgeback
./label_image -i dog.bmp -m **mobilenet_v1_1.0_224_quant.tflite** -l labels.txt 
Loaded model mobilenet_v1_1.0_224_quant.tflite
resolved reporter
invoked 
average time: 85.934 ms 
0.984314: 209 Labrador retriever
0.00392157: 435 bath towel
0.00392157: 208 golden retriever
0.00392157: 169 redbone
0.00392157: 163 beagle
./label_image -i dog.bmp -m **mobilenet_v2_1.0_224.tflite** -l labels.txt 
Loaded model mobilenet_v2_1.0_224.tflite
resolved reporter
invoked 
average time: 180.721 ms 
0.946971: 209 Labrador retriever
0.00624463: 160 Rhodesian ridgeback
0.0035156: 208 golden retriever
0.00168908: 244 bull mastiff
0.0015452: 163 beagle
./label_image -i dog.bmp -m **mobilenet_v2_1.0_224_quant.tflite** -l labels.txt 
Loaded model mobilenet_v2_1.0_224_quant.tflite
resolved reporter
invoked 
average time: 87.001 ms 
**0.780392: 209 Labrador retriever
0.529412: 853 tennis ball
0.529412: 160 Rhodesian ridgeback
0.509804: 208 golden retriever
0.498039: 244 bull mastiff**_

My questionn is that is this a bug about the Mobilenet_V2 quantized models in tensorflow lite 2.0 alpha?  Would you please help? Thanks in advance.

**Comment by JionBin** [comment]
Hi @ravikyram,
Any update? Thanks.

**Comment by freedomtan** [comment]
@JionBin What do you mean by "unreasonable predictions"? The results look fine to me.

**Comment by JionBin** [comment]
@freedomtan For quantized model of Mobilenet_v2, the sum of predictions are not equal to '1'. Is that reasonable?

**Comment by freedomtan** [comment]
@JionBin I figured out what the problem is. The output of the `mobilenet_v2_1.0_224_quant.tflite` is not softmax output! It seems it's something before softmax. If you freeze the graph and convert it to tflite, then you can get expected results. E.g.,
```
freeze_graph --input_graph mobilenet_v2_1.0_224_quant_eval.pbtxt \
--input_checkpoint mobilenet_v2_1.0_224_quant.ckpt \
--output_graph foo_frozen.pb \
--output_node_names=MobilenetV2/Predictions/Reshape_1
```
and
```
tflite_convert --graph_def foo_frozen.pb \
--output_file foo.tflite \
--input_arrays=input \
--output_arrays=MobilenetV2/Predictions/Softmax  \
--input_shape=1,224,224,3 \
--inference_type=QUANTIZED_UINT8 \
--std_dev_values=127 --mean_values=128
```
Then use the quantized MobilenetV2 `foo.tflite` should provide what you expected.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.

**Comment by sanatmpa1** [comment]
@JionBin ,

Can you check @freedomtan 's answer and let us know if it helps in resolving your issue? Thanks!
